onnxruntime/core/providers/cpu/tensor/cast_op.cc

```diff
@@ -170,4 +175,57 @@ Status Cast<MLFloat16>::Compute(OpKernelContext* context) const {
   return st;
 }
 
+ONNX_CPU_OPERATOR_TYPED_KERNEL(
+    Cast,
+    9,
+    string,
+    KernelDefBuilder().TypeConstraint("T1", DataTypeImpl::GetTensorType<std::string>()).TypeConstraint("T2", castOpTypeConstraints),
+    Cast<std::string>);
+
+template <>
+Status Cast<std::string>::Compute(OpKernelContext* context) const {
+  const Tensor* X = context->Input<Tensor>(0);
+  if (X == nullptr) return Status(common::ONNXRUNTIME, common::FAIL, "input count mismatch");
```

Comment:
I think so.. I copied the error message from previous implementation.. will update it.

Reply:
Is is possible to print out operator's name? It's hard for user to debug if there are several Cast's.

---
In reply to: [251634977](https://github.com/Microsoft/onnxruntime/pull/379#discussion_r251634977) [](ancestors = 251634977)

Reply:
well the kernel only takes in an OpKernelContext as parameter, and it does not carry info of the node, which is in opkernel and no methods exposing it in OpKernelContext

Reply:
This code will never be hit. The kernel definition says the input is not optional, so the ONNX checker will validate there's an input. It's only if the input is explicitly marked as optional in the kernel definition that it's possible to see a nullptr here. The ONNX checker will output the op name I believe.

---
In reply to: [251957610](https://github.com/Microsoft/onnxruntime/pull/379#discussion_r251957610) [](ancestors = 251957610)

Reply:
If we remove this line, then GCC's null pointer check will complain X could be NULL. Because GCC doesn't understand ONNX's schema. The nullability of X is not a straightforward. The schema is neither in this file, nor in this project. It's not that obvious.